scripts/globals/baitandswitch.lua

```diff
@@ -0,0 +1,203 @@
+-----------------------------------
+package.loaded["scripts/zones/Metalworks/TextIDs"] = nil;
+-----------------------------------
+require("scripts/zones/Metalworks/TextIDs");
+
+BnS_KILN = 1;
+BnS_REFINERY = 2;
+BnS_REPLICA = 3;
+
+function CheckBaitProgress(player)
+    local scope = 0;
+    local harp = 0;
+    local id = 0;
+    local snares = 0;
+    local charm = 0;
+    local watch = 0;
+    local costume = 0;
+
+    if (player:getMaskBit(player:getVar("Bait_and_Switch_Quest_Progress"),0) == false) then
+        scope = 1;
+    end
+    if (player:getMaskBit(player:getVar("Bait_and_Switch_Quest_Progress"),1) == false) then
+        harp = 1;
+    end
+    if (player:getMaskBit(player:getVar("Bait_and_Switch_Quest_Progress"),2) == false) then
+        id = 1;
+    end
+    if (player:getMaskBit(player:getVar("Bait_and_Switch_Quest_Progress"),3) == false) then
+        snares = 1;
+    end
+    if (player:getMaskBit(player:getVar("Bait_and_Switch_Quest_Progress"),4) == false) then
+        charm = 1;
+    end
+    if (player:getMaskBit(player:getVar("Bait_and_Switch_Quest_Progress"),5) == false) then
+        watch = 1;
+    end
+    if (player:getMaskBit(player:getVar("Bait_and_Switch_Quest_Progress"),6) == false) then
+        costume = 1;
+    end
+
+    local ProgressCheck = {scope,harp,id,snares,charm,watch,costume};
+    return ProgressCheck;
+end;
+
+function CheckSnares(player)
+    local kilnSnare = 0;
+    local shopSnare = 0;
+    local refinerySnare = 0;
+    local replicaSnare = 0;
+    local gunpowderSnare = 0;
+    local cidSnare = 0;
+    local snareNumber = 0;
+
+    if (player:getMaskBit(player:getLocalVar("Bait_and_Switch_Quest_Option"),9) == true) then
+        kilnSnare = 1;
+    end
+    if (player:getMaskBit(player:getLocalVar("Bait_and_Switch_Quest_Option"),10) == true) then
+        shopSnare = 1;
+    end
+    if (player:getMaskBit(player:getLocalVar("Bait_and_Switch_Quest_Option"),11) == true) then
+        refinerySnare = 1;
+    end
+    if (player:getMaskBit(player:getLocalVar("Bait_and_Switch_Quest_Option"),12) == true) then
+        replicaSnare = 1;
+    end
+    if (player:getMaskBit(player:getLocalVar("Bait_and_Switch_Quest_Option"),13) == true) then
+        gunpowderSnare = 1;
+    end
+    if (player:getMaskBit(player:getLocalVar("Bait_and_Switch_Quest_Option"),14) == true) then
+        cidSnare = 1;
+    end
+
+    local SnareCheck = {kilnSnare,shopSnare,refinerySnare,replicaSnare,gunpowderSnare,cidSnare};
+    for i,v in ipairs(SnareCheck) do
+        if (v == 1) then
+            snareNumber = snareNumber + 1;
+        end
+    end
+    local SnareCheckfin = {kilnSnare,shopSnare,refinerySnare,replicaSnare,gunpowderSnare,cidSnare,snareNumber};
+    return SnareCheckfin;
+end;
+
+function RecordBaitProgress(player)
+    local BitConvert = {0,1,2,3,4,5,6};
+    local ChosenItem = CheckOptionForItem(player);
+    local Progress = player:getVar("Bait_and_Switch_Quest_Progress");
```

Comment:
"[quest]BaitAndSwitch" maybe..
meh whatever but at some point we need to redo quests so they're not so annoying to keep track of

Reply:
good idea! we should decide on a standard and start cleaning up that char_vars!

Reply:
esp since there is a sort of max width they will truncate at when I test\*, but somehow when kj checks they work fine. best to not risk weirdness or just scripter mistakes with long varnames anyway.

\* The "ColorHere _RECOMMENDATION_LETTER_USED" tripped the truncation bug for me back when it was added, we shorted it to "ColorHere _R_LETTER_USED" but before that the string was being cut off..ColorHere being one of RED/BLUE/GREEN.